tests/ProxyManagerTest/Functional/LazyLoadingValueHolderFunctionalTest.php

```diff
@@ -375,15 +375,19 @@ static function (?object & $wrappedInstance) : bool {
     }
 
     /**
+     * @template TClass
+     *
+     * @psalm-param (CallableInterface&Mock)|null $initializerMatcher
+     * @psalm-param class-string<TClass> $className
+     * @psalm-param TClass $realInstance
+     *
      * @return Closure(
-     *  object|null,
```

Comment:
`object|null` should be accepted here: the closure is indeed designed to be more permissive